package/MDAnalysis/coordinates/TRC.py

```diff
@@ -224,129 +224,141 @@ def _read_traj_properties(self):
-                                     the wrong number of atoms!")
+                    raise ValueError(
+                        "The trajectory contains the wrong number of atoms!"
+                    )
 
-            elif ("GENBOX" == line.strip()):
+            elif "GENBOX" == line.strip():
                 ntb_setting = int(f.readline())
-                if (ntb_setting == 0):
+                if ntb_setting == 0:
                     frameDat["dimensions"] = None
                     self.periodic = False
 
-                elif (ntb_setting in [-1, 1]):
+                elif ntb_setting in [1, 2]:
                     tmp_a, tmp_b, tmp_c = f.readline().split()
                     tmp_alpha, tmp_beta, tmp_gamma = f.readline().split()
-                    frameDat["dimensions"] = [float(tmp_a),
-                                              float(tmp_b),
-                                              float(tmp_c),
-                                              float(tmp_alpha),
-                                              float(tmp_beta),
-                                              float(tmp_gamma)]
+                    frameDat["dimensions"] = [
+                        float(tmp_a),
+                        float(tmp_b),
+                        float(tmp_c),
+                        float(tmp_alpha),
+                        float(tmp_beta),
+                        float(tmp_gamma),
+                    ]
                     self.periodic = True
 
-                    line3 = f.readline().split()
-                    line4 = f.readline().split()
-                    for v in (line3 + line4):
-                        if (float(v) != 0.0):
-                            raise NotImplementedError("This reader \
-                                                      supports neither \
-                                                      triclinic and/or \
-                                                      (yawed,pitched, \
-                                                      rolled) boxes!")
+                    genbox_line3 = f.readline().split()
+                    if np.count_nonzero(np.array(genbox_line3).astype("float")) != 0:
+                        raise NotImplementedError(
+                            "This reader doesnt't support a shifted origin!"
+                        )
+
+                    genbox_line4 = f.readline().split()
+                    if np.count_nonzero(np.array(genbox_line4).astype("float")) != 0:
+                        raise NotImplementedError(
+                            "This reader "
+                            "doesnt't support "
+                            "yawed, pitched or "
+                            "rolled boxes!"
+                        )
+
                 else:
-                    raise NotImplementedError("This reader does only support\
-                                               vacuum and rectangular boxes!")
+                    raise NotImplementedError(
```

Comment:
Raise a `ValueError` because the problem is that the data (value) is not suitable for the implementation.

`NotImplementedError` has a narrow use in developing classes with inheritance https://docs.python.org/3/library/exceptions.html#NotImplementedError

Reply:
Btw, why are truncated-octahedral PBC not supported? Isn't that just a specific triclinic box?

Reply:
The  truncated-octahedral PBC is one of four boundary conditions supported by the gromos code (see my comment above). You are correct that it can be converted into a triclinic box and gromos has the tools to do that. I didn't want to do some internal conversions hidden from the user in the mda code. Users would need to convert the trajectory from trunc. octa. to triclinic using gromos tools, then use mda.

Reply:
Thank you for pointing out the  narrow use of NotImplementedError!
I changed it to ValueError.